lib/kb_plant_rast/kb_plant_rastImpl.py

```diff
@@ -0,0 +1,170 @@
+            if(SeqLen < 10):
+                output['skipped']+=1
+                continue
+            seq_kmers = [Seq[i:i + Kmer_Length] for i in range(SeqLen-Kmer_Length+1)]
+            for kmer in seq_kmers:
+                if(kmer in Kmers_Functions):
+                    if(ftr['id'] not in Hit_Proteins):
+                        Hit_Proteins[ftr['id']]={}
+                    if(Kmers_Functions[kmer] not in Hit_Proteins[ftr['id']]):
+                        Hit_Proteins[ftr['id']][Kmers_Functions[kmer]]=0
+                    Hit_Proteins[ftr['id']][Kmers_Functions[kmer]]+=1
+
+        #Scan for multi-functional hits
+        #If a function has more hits than others, it takes precendence
+        #If there are more than one function with an equal number of hits, the feature is removed
+        Delete_Proteins={}
+        for ftr in Hit_Proteins.keys():
+            if(len(Hit_Proteins[ftr])==1):
+                continue
+            
+            Top_Hit_Functions={}
+            for function in Hit_Proteins[ftr].keys():
+                if(Hit_Proteins[ftr][function] not in Top_Hit_Functions):
+                    Top_Hit_Functions[Hit_Proteins[ftr][function]]={}
+                Top_Hit_Functions[Hit_Proteins[ftr][function]][function]=1
+
+            Top_Number = (sorted(Top_Hit_Functions.keys(),reverse=True))[0]
+            if(len(Top_Hit_Functions[Top_Number].keys())>1):
+                # delete hit
+                Delete_Proteins[ftr]=1
+                pass
+            else:
+                Top_Function = Top_Hit_Functions[Top_Number].keys()[0]
+                Hit_Proteins[ftr]={Top_Function:Top_Number}
+            
+        #remove the egregious proteins
+        for ftr in Delete_Proteins.keys():
+            del(Hit_Proteins[ftr])
+
+        #count functions
+        Hit_Functions={}
+        for ftr in Hit_Proteins.keys():
+            Hit_Functions[Hit_Proteins[ftr].keys()[0]]=1
+
+        output['hit_ftrs']=len(Hit_Proteins.keys())
+        output['hit_fns']=len(Hit_Functions.keys())
+
+        #Now, re-populate feature functions, and save genome object
+        for ftr in plant_genome['data']['features']:
+            if(ftr['id'] in Hit_Proteins):
+                if('function' in ftr):
+                    old_function = ftr['function']
+                ftr['function'] = Hit_Proteins[ftr['id']].keys()[0]
+        
+        if('output_genome' not in input):
+            input['output_genome']=input['input_genome']
+
+        save_result = wsClient.save_objects({'workspace':input['input_ws'],
+                                             'objects': [{'name' : input['output_genome'],
+                                                          'data' : plant_genome['data'],
+                                                          'type' : "KBaseGenomes.Genome",
+                                                          'meta' : plant_genome['info'][10]}]})
+
+
+        html_string=""
```

Comment:
Shouldn't the method have a text description of the number of features annotated, number of ties ect?